src/rdkafka_transport.c

```diff
@@ -498,6 +500,14 @@ static int rd_kafka_transport_ssl_connect (rd_kafka_broker_t *rkb,
 	if (!SSL_set_fd(rktrans->rktrans_ssl, rktrans->rktrans_s))
 		goto fail;
 
+	mtx_lock(&rkb->rkb_lock);
+	rd_snprintf(name, sizeof(name), "%s", rkb->rkb_nodename);
+	mtx_unlock(&rkb->rkb_lock);
+	if ((t = strrchr(name, ':')))
+		*t = '\0';
+	if (!SSL_set_tlsext_host_name(rktrans->rktrans_ssl, name))
```

Comment:
In most cases, SNI would be not used and ignored by the receiving end. For compilation, the libav stuff is from 2012. In current OpenSSL version, the function itself is a define, so it would be more logical to use `#ifdef SSL_set_tlsext_hostname`. The problem with ifdefs is that they can break. If a newer version of OpenSSL changes the define by an actual function, this won't work any more and you will silently drop the ability to specify SNI. Idem for alternate implementations (LibreSSL, WolfSSL, BoringSSL). Hence my suggestion to check if a define is really needed.